packages/std/src/ado_base/rates.rs

```diff
@@ -160,24 +163,53 @@ impl LocalRate {
         event = event.add_attribute(
             "payment",
             PaymentAttribute {
-                receiver: self.recipient.address.get_raw_address(&deps)?.to_string(),
+                receiver: self
+                    .recipient
+                    .address
+                    .get_raw_address(&deps)
+                    .unwrap_or(Addr::unchecked(self.recipient.address.to_string()))
+                    .to_string(),
                 amount: fee.clone(),
             }
             .to_string(),
         );
-
-        let msg = if is_native {
-            self.recipient
-                .generate_direct_msg(&deps, vec![fee.clone()])?
-        } else {
-            self.recipient.generate_msg_cw20(
-                &deps,
-                Cw20Coin {
-                    amount: fee.amount,
-                    address: fee.denom.to_string(),
+        let msg = if self.recipient.is_cross_chain() {
```

Comment:
We should ensure cw20 isn't used for cross-chain here, maybe at the point of creating the rates recipient we validate?

Reply:
That would work when the `LocalRateValue` is flat, but when it's a percentage we'll have to handle it while processing the rate right?